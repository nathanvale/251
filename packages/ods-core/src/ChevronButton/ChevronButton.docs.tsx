import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { ChevronButtonProps, ChevronButton } from "./ChevronButton";

export const docs: ComponentDocs<ChevronButtonProps> = {
  category: "Interaction",
  componentName: "ChevronButon",
  description: [
    `By default, this component renders a native "button" element. `,
    `If youâ€™d like to customise the implementation of this component`,
    `(e.g. to use a React Router link) you should use ChevronLinkRenderer instead.`,
  ].join(" "),
  propDescriptions: {
    type: "Sets or returns the type of a button",
    domProps:
      "DOM AnchorHTMLAttributes with exception of type, onClick and children",
  },
  migrationGuide: false,

  examples: {
    default: {
      Code: () => <ChevronButton>Chevron button</ChevronButton>,
    },
    additional: [
      {
        label: "Setting the color",
        Code: () => (
          <ChevronButton color="secondaryB">Chevron button</ChevronButton>
        ),
      },
    ],
  },
  snippets: [
    {
      label: "Default",
      Code: () => <ChevronButton>Chevron button</ChevronButton>,
    },
  ],
};
