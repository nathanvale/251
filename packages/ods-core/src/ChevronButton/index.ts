export { ChevronButton } from "./ChevronButton";
export type ChevronButtonProps = import("./ChevronButton").ChevronButtonProps;
