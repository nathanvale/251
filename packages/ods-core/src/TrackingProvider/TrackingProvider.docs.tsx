/* eslint-disable no-alert */
/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs, ExampleDocs } from "@origin-digital/ods-types";
import {
  TrackingProvider,
  TrackingProviderProps,
  Button,
  Card,
  Text,
  TextLink,
} from "..";

export const setupTracking: ExampleDocs = {
  label: "Setting up tracking in your app",
  description: (
    <Text>
      To provide tracking, all you have to do is wrap{" "}
      <TextLink href="#/components/TrackingProvider">TrackingProvider</TextLink>{" "}
      around the root of your app and pass in a handler that expects a tracking
      event object that contains a data-id, type of interaction component (e.g.
      Button), a label (e.g. Button label) and an optional postClickState (e.g.
      An accordion that has opened or closed).
    </Text>
  ),
  noCard: true,
  Code: () => (
    <TrackingProvider
      onTrackingCapture={(props) => {
        alert(
          `data-id: ${props["data-id"]}, type: ${props.type}, label: ${props.label}, postClickState: ${props.postClickState}`
        );
      }}
    >
      <Card>
        <Button>I am being tracked!</Button>
      </Card>
    </TrackingProvider>
  ),
};

export const setupReportingClient: ExampleDocs = {
  label: "Hooking in the @origin-digital/reporting-client",
  playroom: false,
  codeString: `import {
  interaction,
  IInteractionData
} from "@origin-digital/reporting-client";

const App = ()=><TrackingProvider
  onTrackingCapture={({ "data-id": dataId , label, type, postClickState }) => {
    const interactionData: IInteractionData = {
      dataId,
      componentType: type,
      label,
      state: postClickState
    };
    return interaction("click", interactionData);
  }}
>
  <Button>
    I am being tracked by the @origin-digital/reporting-client!
  </Button>
</TrackingProvider>`,
};

export const docs: ComponentDocs<TrackingProviderProps> = {
  category: "Logic",
  componentName: "TrackingProvider",
  description:
    "The TrackingProvider allows use to pass in a tracking handler that captures tracking events from ODS interaction components.",
  propDescriptions: {
    onTrackingCapture:
      "A tracking handler that that expects a TrackingEvent type.",
  },
  migrationGuide: false,
  examples: {
    default: {},
    additional: [setupTracking, setupReportingClient],
  },
  snippets: [],
};
