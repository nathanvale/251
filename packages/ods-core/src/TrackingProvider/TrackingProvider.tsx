import React, { createContext } from "react";
import { TrackingEventHandler } from "@origin-digital/ods-types";

export interface TrackingProviderProps {
  children: React.ReactNode;
  onTrackingCapture: TrackingEventHandler;
}

export const TrackingContext = createContext<
  | undefined
  | {
      onTrackingCapture: TrackingEventHandler;
    }
>(undefined);

export const TrackingProvider = ({
  children,
  onTrackingCapture,
}: TrackingProviderProps) => {
  return (
    <TrackingContext.Provider value={{ onTrackingCapture }}>
      {children}
    </TrackingContext.Provider>
  );
};

TrackingProvider.displayName = "TrackingProvider";
