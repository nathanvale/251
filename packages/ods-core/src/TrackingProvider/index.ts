export { TrackingProvider } from "./TrackingProvider";
export { useTracking } from "./useTracking";

export type TrackingProviderProps = import("./TrackingProvider").TrackingProviderProps;
