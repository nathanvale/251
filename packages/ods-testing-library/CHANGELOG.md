# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
