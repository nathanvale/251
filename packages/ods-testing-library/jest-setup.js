import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
