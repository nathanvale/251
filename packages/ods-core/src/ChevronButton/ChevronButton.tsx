/* eslint-disable jsx-a11y/anchor-has-content */
import React, { AllHTMLAttributes } from "react";
import {
  OptionalTrackableProps,
  ChevronVaraints,
} from "@origin-digital/ods-types";
import {
  ChevronLinkRenderer,
  ChevronLinkRendererProps,
} from "../ChevronLinkRenderer/ChevronLinkRenderer";
import { Text } from "../Text";
import { useTracking } from "../TrackingProvider";

export interface ChevronButtonProps
  extends Omit<ChevronLinkRendererProps, "children" | "variant">,
    OptionalTrackableProps {
  variant?: ChevronVaraints;
  onClick?: AllHTMLAttributes<HTMLButtonElement>["onClick"];
  type?: "button" | "submit" | "reset";
  children?: React.ReactNode;
  domProps?: Omit<
    AllHTMLAttributes<HTMLButtonElement>,
    "onClick" | "children" | "type"
  >;
}
const defaultDataId = "chevron-button";
const defaultVariant = "primary";
export const ChevronButton = ({
  "data-id": dataId = defaultDataId,
  variant = defaultVariant,
  children,
  domProps,
  ...rest
}: ChevronButtonProps) => {
  const { onClickCapture, ref } = useTracking({
    children,
    "data-id": dataId,
    type: ChevronButton.displayName,
  });
  return (
    <ChevronLinkRenderer variant={variant}>
      {({ chevronLinkStyles, IconChevron }) => (
        <button
          data-id={dataId}
          ref={ref}
          onClickCapture={onClickCapture}
          className={chevronLinkStyles}
          {...rest}
          {...domProps}
        >
          <IconChevron />
          <Text>{children}</Text>
        </button>
      )}
    </ChevronLinkRenderer>
  );
};

ChevronButton.defaultProps = {
  "data-id": defaultDataId,
  variant: defaultVariant,
};

ChevronButton.displayName = "ChevronButton";
