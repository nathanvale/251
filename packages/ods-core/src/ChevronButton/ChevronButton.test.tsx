import React from "react";
import { render, fireEvent } from "@origin-digital/ods-testing-library";
import { TrackingEventHandler } from "@origin-digital/ods-types";
import { ChevronButton } from "../ChevronButton/ChevronButton";
import { TrackingProvider } from "../TrackingProvider/TrackingProvider";

test("It can render", () => {
  const { container } = render(<ChevronButton>Chevron link</ChevronButton>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render another tone", () => {
  const { container } = render(
    <ChevronButton variant="secondary">Chevron button</ChevronButton>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it("should call tracking handler when passed into the TrackingProvider", () => {
  const label = "label";

  const onTrackingCapture: TrackingEventHandler = jest.fn();

  const { getByText } = render(
    <TrackingProvider onTrackingCapture={onTrackingCapture}>
      <ChevronButton>{label}</ChevronButton>
    </TrackingProvider>
  );
  const node = getByText(label);
  fireEvent.click(node);
  expect(onTrackingCapture).toHaveBeenCalledWith({
    "data-id": "chevron-button",
    type: "ChevronButton",
    label,
    postClickState: undefined,
  });
});
