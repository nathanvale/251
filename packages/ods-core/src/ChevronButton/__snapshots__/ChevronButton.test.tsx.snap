// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It can render 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  max-width: 100%;
  text-align: left;
}

<button
  class="typography-linkReset-1 typography-chevron-link-2"
  data-id="chevron-button"
>
  <div
    class="c0"
    style="margin-left: -8px; margin-top: 0px; margin-bottom: 0px; height: 24px;"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root svg-icon-secondary-small-svg-icon-3"
      data-id="IconChevronRight"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </div>
  <span
    class="typography-body-variant-basekick-13 typography-neutral-color-14 typography-regular-weight-15 typography-strong-support-16 -span c1"
    data-id="text"
  >
    Chevron link
  </span>
</button>
`;

exports[`It can render another tone 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  max-width: 100%;
  text-align: left;
}

<button
  class="typography-linkReset-18 typography-chevron-link-19"
  data-id="chevron-button"
>
  <div
    class="c0"
    style="margin-left: -8px; margin-top: 0px; margin-bottom: 0px; height: 24px;"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root svg-icon-secondary-small-svg-icon-20"
      data-id="IconChevronRight"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </div>
  <span
    class="typography-body-variant-basekick-30 typography-neutral-color-31 typography-regular-weight-32 typography-strong-support-33 -span c1"
    data-id="text"
  >
    Chevron button
  </span>
</button>
`;
